icu4c/source/common/brkeng.cpp

```diff
@@ -128,41 +131,45 @@ static void U_CALLCONV _deleteEngine(void *obj) {
 U_CDECL_END
 U_NAMESPACE_BEGIN
 
-const LanguageBreakEngine *
-ICULanguageBreakFactory::getEngineFor(UChar32 c) {
-    const LanguageBreakEngine *lbe = nullptr;
-    UErrorCode  status = U_ZERO_ERROR;
-
+void ICULanguageBreakFactory::ensureEngines(UErrorCode& status) {
     static UMutex gBreakEngineMutex;
     Mutex m(&gBreakEngineMutex);
-
     if (fEngines == nullptr) {
         LocalPointer<UStack>  engines(new UStack(_deleteEngine, nullptr, status), status);
```

Comment:
You could completely remove the `_deleteEngine()` function by making `LanguageBreakEngine` derive from `UObject` and using `uprv_deleteUObject()`